### ISSUE TITLE
Colons were allowed in v5.0.0 but disallowed in v5.1.1 making it difficult to upgrade existing stacks

### ISSUE THREAD

**Issue by vazexqi** [pulumi/pulumi-cloudflare#489]
### What happened?

I am using the automation API but I don't think this is affected by that.
I am using pulumi with my own store. In production I used Google Cloud Storage but this issue can be replicated easily with a local file storage as well.

In v5.0.0 of this provider, I created a resource with a name that uses colons. For better or worse, this naming convention seemed like a good idea at the time and it was also accepted by the pulumi engine.

When I upgraded to v5.1.1 (and above), I got an error that the name was not valid.
This is causing issues because I don't have a way to cleanly upgrade `@pulumi/cloudflare` without renaming all my resources (I have hundreds of them).

Works before in v5.0.0 and saves to state.
```
      const dns = new DNSRecord("somename::anothername", {
        zoneId: process.env.CLOUDFLARE_ZONE_ID!,
        name: "nick-subdomain",
        value: "example.com",
        type: "CNAME",
        proxied: true,
      });
```

No longer works in v5.1.1. and above (and throws an error)

```
      const dns = new DNSRecord("somename::anothername", {
        zoneId: process.env.CLOUDFLARE_ZONE_ID!,
        name: "nick-subdomain",
        value: "example.com",
        type: "CNAME",
        proxied: true,
      });

Error:   cloudflare:index:Record (somename):
    error: URN 'urn:pulumi:naming-issue::bug::cloudflare:index/record:Record::somename::anothername' is not valid
```

### Example

```
import "dotenv/config";

import { LocalWorkspace } from "@pulumi/pulumi/automation/index.js";
import { Record as DNSRecord } from "@pulumi/cloudflare";

const stack = await LocalWorkspace.createOrSelectStack(
  {
    projectName: "bug",
    stackName: "naming-issue",
    program: async () => {
      const dns = new DNSRecord("somename::anothername", {
        zoneId: process.env.CLOUDFLARE_ZONE_ID!,
        name: "nick-subdomain",
        value: "example.com",
        type: "CNAME",
        proxied: true,
      });
    },
  },
  {
    envVars: {
      CLOUDFLARE_API_TOKEN: process.env.CLOUDFLARE_API_TOKEN!,
    },
  }
);
const upRes = await stack.up({
  onOutput: console.info,
});

console.dir(upRes);
```

### Output of `pulumi about`

```
➜ pulumi about
CLI          
Version      3.62.0
Go Version   go1.20.2
Go Compiler  gc

Plugins
NAME        VERSION
cloudflare  5.1.1
nodejs      unknown

Host     
OS       darwin
Version  13.5.2
Arch     arm64

... omitted

Dependencies:
NAME                VERSION
@types/node         18.17.16
dotenv              16.3.1
typescript          5.2.2
@pulumi/cloudflare  5.1.1
@pulumi/pulumi      3.82.1
```

### Additional context

How can I upgrade to a newer version of @pulumi/cloudflare?
Is the only workaround to manipulate the state?

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mikhailshilkov** [comment]
This error is actually coming from the bridge muxer: https://github.com/pulumi/pulumi-terraform-bridge/blob/7e6c8fe7c8cd72dd6a5099fe932b290d8ad67ed9/x/muxer/muxer.go#L297 which relies on `IsValid` as mentioned in https://github.com/pulumi/pulumi/issues/13968#issue-1899180884

I think it's a dupe of https://github.com/pulumi/pulumi/issues/13968 but I'll see what they respond on it.

**Comment by mikhailshilkov** [comment]
The upstream issue has been closed, we need to wait until the bridge updates to it and then we can close this issue too

**Comment by vazexqi** [comment]
@mikhailshilkov - How do I track if the bridge has been updated?

**Comment by vazexqi** [comment]
We can closed this now. Just had a chance to test this end-to-end.

I managed to trace back the releases.

The upstream fix in https://github.com/pulumi/pulumi/pull/14107 was released in v3.95.0
The bridge incorporated the fix in https://github.com/pulumi/pulumi-terraform-bridge/releases/tag/v3.67.0 (Dec 5, 2023)
And finally this repo incorporated the changes in https://github.com/pulumi/pulumi-cloudflare/releases/tag/v5.17.0 (Dec 27, 2023)
